src/Randomizer.SMZ3.Twitch/TwitchChatClient.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+
+using Microsoft.Extensions.Logging;
+
+using Randomizer.SMZ3.ChatIntegration;
+
+using TwitchLib.Client;
+using TwitchLib.Client.Models;
+
+namespace Randomizer.SMZ3.Twitch
+{
+    public class TwitchChatClient : IChatClient
+    {
+        private readonly TwitchClient _twitchClient;
+
+        public TwitchChatClient(ILogger<TwitchChatClient> logger)
+        {
+            Logger = logger;
+
+            _twitchClient = new();
+            _twitchClient.OnConnected += _twitchClient_OnConnected;
+            _twitchClient.OnDisconnected += _twitchClient_OnDisconnected;
+            _twitchClient.OnMessageReceived += _twitchClient_OnMessageReceived;
+        }
+
+        public event EventHandler? Connected;
+
+        public event MessageReceivedEventHandler? MessageReceived;
+
+        public string? ConnectedAs { get; protected set; }
+
+        public string? Channel { get; protected set; }
+
+        public bool IsConnected { get; protected set; }
+
+        protected ILogger<TwitchChatClient> Logger { get; }
+
+        public void Connect(string userName, string oauthToken, string channel)
+        {
+            var credentials = new ConnectionCredentials(userName, oauthToken);
```

Comment:
Does this work okay if it tries to connect a second time? For example, if they close out of the tracker and relaunch it?

Reply:
I'll test that along with bad authentication and such.